### DISCUSSION TITLE
Are SSG and SSR props secure ?

### DISCUSSION THREAD

**Discussion by asotog** [vercel/next.js#51245] [Help] [answered]
### Summary

Hi all,

every time we return props from getServerSideProps or getStaticProps those props objects are printed in the page document, is this expected behavior? 

 asking because we are passing some sensitive data probable 

### Example

<img width="511" alt="image" src="https://github.com/vercel/next.js/assets/813023/bc0a84dc-0f45-4313-b1c0-f8f6cf123c74">

**Comment by icyJoseph** [comment]
Yeah it is expected, since it is the same data you should be using in your UI, but it might backfire.

For example, I had made a GitHub query for all of my repositories, even private, and in the UI I'd filter them out. Guess what, all of them were in the `__NEXT_DATA__` tag anyway. So I had to filter and prep my data within GSP and GSSP and only pass along those things that are needed.

**Comment by asotog** [comment]
thanks @icyJoseph but that means I can not put anything sensitive on props ?

**Reply by icyJoseph** [reply, marked as answer]
I mean, putting things in props makes them available client side. The app Directory solves this with server components, which are rendered server side and never send any JS or unused data to the client.

The problem is that, in pages dir, something called hydration has to happen to the client side and that requires the props that were used server side.
